source/storing-data/archive-versions.rst

```diff
@@ -0,0 +1,240 @@
+(unless the action type is ``import``),
+assigning it the output name registered to the relevant action.
+Prior to this change, if one action returned multiple artifacts of the same :term:`Semantic Type`,
+it was not possible to differentiate between them using provenance alone.
+
+In `commit e072706 <https://github.com/qiime2/qiime2/commit/e07270659bd4089afdeef744868f8b7ac9c698a9>`_,
+it adds provenance support for :term:`Pipelines <Pipeline>`,
+adding the ``alias-of`` key to the ``action`` section of ``action.yaml``.
+See description in :ref:`action-block` for details.
+
+Version 3
+---------
+
+Released in QIIME 2 version `2017.12 <https://github.com/qiime2/qiime2/releases/tag/2017.12.0>`_
+(`changelog <https://forum.qiime2.org/t/qiime-2-2017-12-release-is-now-live/2308>`__),
+`commit 684b8b7 <https://github.com/qiime2/qiime2/commit/684b8b77276bd8fc7228042c674a166669bb5bef>`_,
+the directory structure of this format is identical to v1 and v2.
+
+With this release, QIIME 2 Actions are able to take variadic arguments,
+allowing users to pass collections of Artifacts (``List`` s and ``Set`` s).
+A YAML representer has been added so that ``action.yaml`` can represent ``Set`` s of Artifact inputs.
+These will show up in ``action.yaml`` as custom ``!set`` tags.
+
+Version 4
+---------
+
+Released in QIIME 2 version `2018.4 <https://github.com/qiime2/qiime2/releases/tag/2018.4.0>`_
+(`changelog <https://forum.qiime2.org/t/qiime-2-2018-4-release-is-now-live/3946>`__),
+`commit 00a294c <https://github.com/qiime2/qiime2/commit/00a294cbfc8737c1a2f57d695615098dc9beb4ad>`_,
+this format adds citations to the directory format,
+adds a ``transformers`` section to ``action.yaml``,
+and aligns the structure of ``environment:framework`` (also in ``action.yaml``)
+to match the structure of ``environment::plugins::<some_plugin>``.
+
+Whenever an Action is run, its registered citations are captured.
+When saved, they are written to a ``citations.bib`` file
+inside the Archive's ``provenance`` directory.
+Citations for all of the current Result's ancestors are stored in their respective <UUID> directories
+(e.g. ``/<root_UUID>/provenance/artifacts/<ancestor_UUID>/citations.bib``).
+
+.. figure:: ../img/archive_versions/v4_prov_files.svg
+   :alt: Box and Arrow diagram of the provenance files in a v4 archive, as described above.
+
+**The overall directory structure remains identical to a v1 archive, above.**
+
+Result-specific citation tags are also written to
+the ``transformers`` and ``environment`` sections of the ``action.yaml`` files,
+for the current Result and for all ancestors with registered citations.
+A new custom ``!cite '<citation key>'`` tag is use to support this in YAML.
+
+A ``transformers`` section is added between the ``action`` and ``environment`` sections of ``action.yaml``.
+Because Pipelines do not use transformers,
+transformers will be recorded only for :term:`Methods <Method>` and :term:`Visualizers <Visualizer>`.
```

Comment:
Also import!

Reply:
Good catch. Does this feel about right? 
![image](https://user-images.githubusercontent.com/39198770/126391501-44a01fd2-08fc-454d-ac9c-649994d6b89a.png)

If so, I think we're good to go.

Reply:
yep!

Reply:
Actually perhaps make it `and when imporing data (such as with qiime tools import)`, since you can import in arbitrary interfaces.